application_form/services/reservation.py

```diff
@@ -88,11 +104,31 @@ def create_reservation_without_application(
         if user:
             reservation_data["handler"] = user.profile_or_user_full_name
 
+        ownership_type = apartment.project_ownership_type
+        if ownership_type.lower() == "haso":
+            existing_reservations = existing_reservations.order_by("queue_position")
+            existing_reservations = existing_reservations.active()
+            right_of_residence = reservation_data["customer"].right_of_residence
+
+            right_of_residence_ordering_number = get_residence_ordering_number(
+                right_of_residence
+            )
```

Comment:
Is it always new batch now? I thought that it's still possible to have a number from the old batch, but I'm not sure. Is there a value in `reservation_data` for "is_old_batch" (don't know the actual spelling)?

Reply:
Actually, isn't there already the property in the customer model for this?

Maybe this could be replaced with:
```python
customer = reservation_data["customer"]
right_of_residence_ordering_number =  customer.right_of_residence_ordering_number
```
If that property really is available here, then there's no need to move the code to another function either.

Reply:
This was only available in the Application model, and since this is a reservation without application, I couldn't access it, that's why I moved it

Reply:
To me it seems like it could be available through customer too, since the Customer class inherits from CommonApplicationData which implements the property.